[SDK-4218] Flutter Web quickstart (#10265)
auth0/docs: articles/quickstart/spa/flutter/01-login.md

```dart
auth0.onLoad().then((final credentials) => setState(() {
    // Handle or store credentials here
    _credentials = credentials;
  }));
```

::::checkpoint
:::checkpoint-default
Add a button to your app that calls `loginWithRedirect()` and logs the user into your app. Verify that you are redirected to Auth0 for authentication and then back to your application.

Verify that you can access `credentials` as a result of calling `onLoad` and that you're able to access the ID and access tokens.
:::

:::checkpoint-failure
If your application did not launch successfully:

- Ensure the Allowed Callback URLs are set properly
- Verify you saved your changes after entering your URLs
- Make sure the domain and client ID values are imported correctly

Still having issues? Check out our [documentation](https://auth0.com/docs) or visit our [community page](https://community.auth0.com) to get more help.
:::
::::

## Add logout to your app

To log users out, redirect them to the Auth0 logout endpoint to clear their login session by calling the Auth0 Flutter SDK `logout()`. [Read more about logging out of Auth0](https://auth0.com/docs/authenticate/login/logout).

> See this example of an `ElevatedButton` widget that logs the user out of the app. Note that `_credentials` is set to `null`, indicating that the user is no longer signed in to your app:

Reviewer: ```suggestion
See this example of an `ElevatedButton` widget that logs the user out of the app:
```

  Credentials? _credentials;

  late Auth0Web auth0;

  @override
  void initState() {
    super.initState();
    auth0 = Auth0Web('${account.namespace}', '${account.clientId}');
  }

  @override
  Widget build(BuildContext context) {
    // ...
  }
}
```

Next, redirect your users to the Auth0 Universal Login page using `loginWithRedirect`. See this example of a `ElevatedButton` widget that logs the user in when clicked:

:::note
You will normally need to specify the `redirectUrl` parameter to `loginWithRedirect`. Omitting this will cause Auth0 to use the [default login route](https://auth0.com/docs/authenticate/login/auth0-universal-login/configure-default-login-routes), which is not configured by default.
:::

```dart
if (_credentials == null) {
  ElevatedButton(
    onPressed: () => auth0.loginWithRedirect(redirectUrl: 'http://localhost:3000'),
    child: const Text("Log in")
  )
}
```

When a user logs in, they are redirected back to your application. You are then able to access the ID and access tokens for this user by calling `onLoad` during startup and handling the credentials that are given to you:

```dart
auth0.onLoad().then((final credentials) => setState(() {

>     // handle or store credentials here

Reviewer: ```suggestion
    // Handle or store credentials here
    _credentials = credentials;
```
Let's make it complete

  }));
```

::::checkpoint
:::checkpoint-default
Add a button to your app that calls `loginWithRedirect()` and logs the user into your app. Verify that you are redirected to Auth0 for authentication and then back to your application.

Verify that you can access `credentials` as a result of calling `onLoad` and that you're able to access the ID and access tokens.
:::

:::checkpoint-failure
If your application did not launch successfully:

> - Ensure you set the Allowed Callback URLs are correct

Reviewer: ```suggestion
- Ensure the Allowed Callback URLs are set properly
```

- Verify you saved your changes after entering your URLs
- Make sure the domain and client ID values are imported correctly

Still having issues? Check out our [documentation](https://auth0.com/docs) or visit our [community page](https://community.auth0.com) to get more help.
:::
::::

## Add logout to your app

To log users out, redirect them to the Auth0 logout endpoint to clear their login session by calling the Auth0 Flutter SDK `logout()`. [Read more about logging out of Auth0](https://auth0.com/docs/authenticate/login/logout).

See this example of an `ElevatedButton` widget that logs the user out of the app. Note that `_credentials` is set to `null`, indicating that the user is no longer signed in to your app:

```dart
ElevatedButton(
  onPressed: () async {

>     setState(() {
>       _credentials = null;
>     });
> 

Reviewer: ```suggestion
```
I think this is not actually necessary, as after the redirect onLoad() will kick in. Will remove it from the sample app as well (will leave it only for mobile).

Author: I agree

The user profile automatically retrieves user profile properties for you when the page loads, and can be accessed and stored by calling `onLoad` during application startup. The returned object from `onLoad` contains a `user` property with all the user profile properties. This is internally populated by decoding the ID token.

See this example of a `View` component that displays the user profile on the screen:

```dart
import 'package:auth0_flutter/auth0_flutter.dart';
import 'package:flutter/material.dart';

class ProfileView extends StatelessWidget {
  const ProfileView({Key? key, required this.user}) : super(key: key);

  final UserProfile user;

  @override
  Widget build(BuildContext context) {
    return Column(
      mainAxisAlignment: MainAxisAlignment.center,
      children: [
        if (user.name != null) Text(user.name!),
        if (user.email != null) Text(user.email!)
      ],
    );
  }
}
```

::::checkpoint
:::checkpoint-default
Log in and inspect the `user` property on the result. Verify the current user's profile information, such as `email` or `name`.
:::
:::checkpoint-failure
If your application did not return user profile information:

> - Verify the access token is valid

Reviewer: ```suggestion
- Verify the ID token is valid
```
